src/main.py

```diff
@@ -284,34 +284,48 @@ def _run_testing(env: BaseEnv, approach: BaseApproach) -> Metrics:
         else:
             monitor = None
         try:
-            if CFG.env == "behavior" and \
-                CFG.behavior_option_model_eval:  # pragma: no cover
-                # To evaluate BEHAVIOR on our option model, we are going
-                # to run our approach's plan on our option model.
-                # Note that if approach is not a BilevelPlanningApproach
-                # we cannot use this method to evaluate and would need to
-                # run the policy on the option model, not the plan
-                assert isinstance(approach, BilevelPlanningApproach)
-                last_plan = approach.get_last_plan()
-                option_model_start_time = time.time()
-                traj, solved = _run_plan_with_option_model(
-                    task, test_task_idx, approach.get_option_model(),
-                    last_plan)
-                execution_metrics = {
-                    "policy_call_time": option_model_start_time - time.time()
-                }
+            # Will evaluate all approaches on our simulator with the options
+            # to (1) only evalutate based on ability to find a plan for
+            # BilevelPlanningApproaches and (2) for BEHAVIOR to evaluate on
+            # option models instead of the low-level simulator (with the
+            # plan_only_eval and behavior_option_model_eval flags, respectively)
+            if CFG.plan_only_eval and isinstance(
+                    approach, BilevelPlanningApproach):  # pragma: no cover
```

Comment:
if this is not behavior-specific, we should add a test for it. I'm not sure if we will want to keep it after we factor out all of the behavior code into a new branch, though. maybe @ronuchit has thoughts

Reply:
i guess i'll look after the other comments are addressed? but yeah just at a glance of these lines, they're hacky and we should find a better way to do it

Reply:
Moved to different PR